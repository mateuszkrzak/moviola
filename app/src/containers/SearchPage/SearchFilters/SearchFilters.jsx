import React from 'react';

import './SearchFilters.scss';

class SearchFilters extends React.Component {
  state = {
    selectedFilter: 'title',
  };

  clickHandler = (e) => {
    const selectedFilter = e.target.innerHTML.toLowerCase();

    this.setState({ selectedFilter }, () => this.props.onChange(this.state.selectedFilter));
  };

  isActive(filter) {
    return this.state.selectedFilter === filter ? 'active' : '';
  }

  static getDerivedStateFromProps(props, state) {
    return {
      selectedFilter: props.value === 'genres' ? 'genres' : 'title',
    }
  }

  render() {
    return (
      <div styleName="wrapper">
        <span styleName="title">Search by</span>
        <button
          onClick={this.clickHandler}
          className="standard-button"
          styleName={this.isActive('title')}
        >
          Title
        </button>
        <button
          onClick={this.clickHandler}
          className="standard-button"
          styleName={this.isActive('genres')}
        >
          Genres
        </button>
      </div>
    );
  }
}
export default SearchFilters;
