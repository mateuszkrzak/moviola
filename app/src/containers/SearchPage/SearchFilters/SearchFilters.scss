@import "../../../common/styles/variables.scss";

.wrapper {
  display: inline;

  .standard-button:last-child {
    margin-right: 0;
  }
}

.title {
  text-transform: uppercase;
  margin-right: 15px;
  font-weight: 500;
}

.active {
  background: $pink;
}
