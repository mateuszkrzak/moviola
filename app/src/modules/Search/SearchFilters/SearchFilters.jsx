import React from 'react';
import CSSModules from 'react-css-modules';

import styles from './SearchFilters.scss';

const SearchFilters = () => (
  <div styleName="wrapper">
    <span styleName="title">Search by</span>
    <button className="standard-button">Title</button>
    <button className="standard-button" styleName="active">Genre</button>
  </div>
);
export default CSSModules(SearchFilters, styles);
