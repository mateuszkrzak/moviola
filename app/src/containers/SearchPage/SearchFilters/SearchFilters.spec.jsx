import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchFilters from './SearchFilters';

describe('SearchFilters component ', () => {
  test('should call on change handler when click on filters', () => {
    const spy = jest.fn();
    const component = shallow(<SearchFilters value='testValue' onChange={spy} />);
    const titleButton = component.find('button').at(0);
    const genreButton = component.find('button').at(1);
    
    genreButton.simulate('click', { target: { innerHTML: 'genres' }});
    titleButton.simulate('click', { target: { innerHTML: 'title' }});

    expect(spy.mock.calls).toMatchSnapshot();
  });
});
