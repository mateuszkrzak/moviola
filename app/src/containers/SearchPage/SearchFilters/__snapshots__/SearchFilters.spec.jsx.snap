// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilters component  should call on change handler when click on filters 1`] = `
Array [
  Array [
    "genres",
  ],
  Array [
    "title",
  ],
]
`;
